workflows/gpu_operator_dashboard/fetch_ci_data.py

```
@@ -452,69 +471,175 @@ def process_closed_prs(results_by_ocp: Dict[str, List[Dict[str, Any]]]) -> None:
+
+    # Process new results (should already be filtered, but apply for safety)
+    for item in new_tests:
+        result = TestResult(**item)
+        # Filter: only include results with exact semantic versions and not ABORTED
+        if result.has_exact_versions() and result.test_status != STATUS_ABORTED:
+            version_key = _get_version_key(result)
+            results_by_version.setdefault(version_key, []).append(result)
+
+    # Keep exactly one result per version key
+    final_results = []
+    for version_key, version_results in results_by_version.values():
```

Ruff (0.12.2)

Loop control variable `version_key` not used within loop body

(B007)